sway-core/src/semantic_analysis/ast_node/expression/match_expression/typed/typed_scrutinee.rs

```diff
@@ -104,6 +104,44 @@ impl ty::TyScrutinee {
             Scrutinee::Error { err, .. } => Err(err),
         }
     }
+
+    /// Returns true if the `TyScrutinee` consists only of catch-all scrutinee variants, recursively.
+    /// Catch-all varaints are .., _, and variables.
+    /// E.g.: (_, x, Point { .. })
+    /// A catch-all scrutinee matches all the values of the corresponding type.
+    pub(crate) fn is_catch_all(&self) -> bool {
+        match &self.variant {
+            ty::TyScrutineeVariant::CatchAll => true,
+            ty::TyScrutineeVariant::Variable(_) => true,
+            ty::TyScrutineeVariant::Literal(_) => false,
+            ty::TyScrutineeVariant::Constant { .. } => false,
+            ty::TyScrutineeVariant::StructScrutinee { fields, .. } => {
+                let mut is_catch_all_struct_pattern = true;
```

Comment:
Is this the same as

```rust
fields.iter().filter_map(|x| x.scrutinee.as_ref()).all(|x| x.is_catch_all())
```

Reply:
Thanks a lot for this, the code is so much simpler and nicer now! I originally conveniently took over the whole matching code from [here](https://github.com/FuelLabs/sway/blob/04a597093e7441898933dd412b8e4dc6ac860cd3/sway-core/src/semantic_analysis/ast_node/expression/match_expression/analysis/pattern.rs#L119) and then pretty brainlessly got caught in the original structure.